test/gtest/shp/transform.cpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+#include "shp-tests.hpp"
+
+namespace drtesting {
+std::optional<bool> expect_fast_transform_path;
+}
+
+#include "dr/shp/algorithms/transform.hpp"
+
+template <typename AllocT> class TransformTest : public testing::Test {
+public:
+  using DistVec = shp::distributed_vector<typename AllocT::value_type, AllocT>;
+  using LocalVec = std::vector<typename AllocT::value_type>;
+  constexpr static const auto add_10_func = std::bind(
+      std::plus<typename AllocT::value_type>(), std::placeholders::_1, 10);
```

Comment:
You must have tried...

Why not something simpler like functor, template function/lambda?